File: frontend/src/types/incident.ts
export interface Incident {
  id: number;
  event_id: number;
  incident_type: string;
  description: string;
  location?: string;
  reported_by: string;
  priority: string;
  assigned_staff?: string;
  status: string;
  reported_at: string;
}

export interface IncidentCreate {
  event_id: number;
  incident_type: string;
  description: string;
  location?: string;
  reported_by: string;
  priority: string;
  assigned_staff?: string;
}

File: backend/app/schemas/incident.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class IncidentCreate(BaseModel):
    event_id: int
    incident_type: str
    description: str
    location: Optional[str] = None
    reported_by: str
    priority: str
    assigned_staff: Optional[str] = None

class IncidentResponse(BaseModel):
    id: int
    event_id: int
    incident_type: str
    description: str
    location: Optional[str] = None
    reported_by: str
    priority: str
    assigned_staff: Optional[str] = None
    status: str
    reported_at: datetime

    class Config:
        from_attributes = True